subprojects/docs/src/docs/release/notes.md

```diff
@@ -141,6 +141,35 @@ In the same vein, doing file system operations such as `copy()`, `sync()` and `d
 
 See the [user manual](userguide/custom_gradle_types.html#service_injection) for how to inject services and the [`FileSystemOperations`](javadoc/org/gradle/api/file/FileSystemOperations.html) and [`ExecOperations`](javadoc/org/gradle/process/ExecOperations.html) api documentation for more details and examples.
 
+### Services available to Worker API actions
+
+The following services are now available for injection in `WorkAction` classes:
+- [ObjectFactory](javadoc/org/gradle/api/model/ObjectFactory.html)
+- [FileSystemOperations](javadoc/org/gradle/api/file/FileSystemOperations.html)
+- [ExecOperations](javadoc/org/gradle/process/Execoperations.html)
+
+These services can be injected by adding them to the constructor arguments of the `WorkAction` implementation:
```

Comment:
❓ Does injecting through methods also works?
```
@Inject
protected FileSystemOperations getFileSystemOperations() {
    throw new UnsupportedOperationException()
}
```

Reply:
It does and it is discussed in the user guide manual section on "Service Injection" but the constructor injection is generally what we would recommend to users.  We would recommend the getter injection in situations where a high degree of inheritance created some awkwardness in passing constructor arguments up the chain.  This doesn't seem likely to be the case in most `WorkAction` implementations I would think.  I think I'd prefer to advertise the constructor injection in the release notes and just leave mention of getter injection in the service injection section of the user guide.